src/elements/content-sidebar/DocgenSidebar.js

```diff
@@ -0,0 +1,86 @@
+// @ts-check
+/**
+ * @flow
+ * @file Sign sidebar component
+ * @author Box
+ */
+
+import flow from 'lodash/flow';
+import * as React from 'react';
+import type { ErrorContextProps } from '../../common/types/api';
+import type { WithLoggerProps } from '../../common/types/logging';
+import { ORIGIN_METADATA_SIDEBAR, SIDEBAR_VIEW_METADATA } from '../../constants';
+import { mark } from '../../utils/performance';
+import { withAPIContext } from '../common/api-context';
+import { withErrorBoundary } from '../common/error-boundary';
+import { withLogger } from '../common/logger';
+import { EVENT_JS_READY } from '../common/logger/constants';
+import SidebarContent from './SidebarContent';
+
+type EUAProps = {
+    enabled: boolean,
+    getDocgenTags: Function,
+};
+
+type ExternalProps = {
+    // isFeatureEnabled: boolean,
+    docgenPreviewSidebarProps: EUAProps,
+};
+
+type DocgenTag = {
+    json_paths: Array<string>,
+    tag_content: string,
+    tag_type: string,
+};
+
+type Props = {
+    isLoading: boolean,
+} & ExternalProps &
+    ErrorContextProps &
+    WithLoggerProps;
+
+type State = { tags: Array<DocgenTag> };
+
+const MARK_NAME_JS_READY = `${ORIGIN_METADATA_SIDEBAR}_${EVENT_JS_READY}`;
+
+mark(MARK_NAME_JS_READY);
+
+// TO DO: implement actual sidebar content in separate tickets
+class DocgenSidebar extends React.PureComponent<Props, State> {
```

Comment:
same comment about abbreviation if this should be `DocGenSidebar` and similar places